views/user.go

```diff
@@ -0,0 +1,402 @@
+)
+
+type (
+	// UsersTemplate represents the context for the user template
+	UsersTemplate struct {
+		Users                                 []UserStripped
+		UserID                                int
+		CurPage, NextPage, PrevPage, LastPage int
+		ActivePage                            string
+		Sort                                  struct {
+			Pages      int
+			Size       int
+			PageNumber int
+			Column     string
+			Direction  string
+			Search     string
+		}
+	}
+)
+
+// UsersFunc handles a users request
+func (v *Views) UsersFunc(w http.ResponseWriter, r *http.Request) {
+	session, _ := v.cookie.Get(r, v.conf.SessionCookieName)
+
+	c := v.getData(session)
+
+	var err error
+
+	if r.Method == "POST" {
+		err = r.ParseForm()
+		if err != nil {
+			err = v.errorHandle(w, err)
+			if err != nil {
+				log.Println(err)
+				return
+			}
+		}
+
+		column := r.FormValue("column")
+		direction := r.FormValue("direction")
+		search := r.FormValue("search")
+		var size int
+		sizeRaw := r.FormValue("size")
+		if sizeRaw == "all" {
+			size = 0
+		} else {
+			size, err = strconv.Atoi(sizeRaw)
+			if err != nil {
+				size = 0
+			} else if size <= 0 {
+				err = v.errorHandle(w, fmt.Errorf("invalid size, must be positive"))
+				if err != nil {
+					log.Println(err)
+					return
+				}
+			} else if size != 5 && size != 10 && size != 25 && size != 50 && size != 75 && size != 100 {
+				size = 25
+			}
+		}
+		valid := true
+		switch column {
+		case "userId":
+		case "name":
+		case "username":
+		case "email":
+		case "lastLogin":
+			break
+		default:
+			valid = false
+		}
+		switch direction {
+		case "asc":
+		case "desc":
+			break
+		default:
+			valid = false
+		}
+		r.Method = "GET"
+		if valid {
+			if len(search) > 0 {
+				if size > 0 {
+					http.Redirect(w, r, fmt.Sprintf("/internal/users?column=%s&direction=%s&search=%s&size=%d&page=1", column, direction, url.QueryEscape(search), size), http.StatusFound)
+				} else {
+					http.Redirect(w, r, fmt.Sprintf("/internal/users?column=%s&direction=%s&search=%s", column, direction, url.QueryEscape(search)), http.StatusFound)
+				}
+			} else {
+				if size > 0 {
+					http.Redirect(w, r, fmt.Sprintf("/internal/users?column=%s&direction=%s&size=%d&page=1", column, direction, size), http.StatusFound)
+				} else {
+					http.Redirect(w, r, fmt.Sprintf("/internal/users?column=%s&direction=%s", column, direction), http.StatusFound)
+				}
+			}
+		} else if len(search) > 0 {
+			if size > 0 {
+				http.Redirect(w, r, fmt.Sprintf("/internal/users?search=%s&size=%d&page=1", url.QueryEscape(search), size), http.StatusFound)
+			} else {
+				http.Redirect(w, r, fmt.Sprintf("/internal/users?search=%s", url.QueryEscape(search)), http.StatusFound)
+			}
+		} else {
+			if size > 0 {
+				http.Redirect(w, r, fmt.Sprintf("/internal/users?size=%d&page=1", size), http.StatusFound)
+			} else {
+				r.URL.Query().Del("*")
+				http.Redirect(w, r, "/internal/users", http.StatusFound)
+			}
+		}
```

Comment:
This is quite hard to understand - do you think it'd be worth changing this to something like

```go
var params url.Values // https://pkg.go.dev/net/url#Values
if len(search) > 0 {
  params.Set("search", search)
}
if valid {
  params.Set("column", column)
}
// ...
http.Redirect(w, r, fmt.Sprintf("/internal/users?%s", params.Encode())
```

Reply:
This is changed later on to have a query builder